pilot/pkg/proxy/envoy/watcher_test.go

```diff
@@ -107,12 +107,20 @@ func TestWatchCerts_Multiple(t *testing.T) {
 }
 
 func TestWatchCerts(t *testing.T) {
-	name, err := ioutil.TempDir(os.TempDir(), "certs")
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "certs")
 	if err != nil {
 		t.Errorf("failed to create a temp dir: %v", err)
 	}
+	// create a temp file
+	tmpFile, err := ioutil.TempFile(tmpDir, "test.file")
+	if err != nil {
+		t.Errorf("failed to create a temp file in testdata/certs: %v", err)
```

Comment:
This function uses `t.Error` throughout, rather than `t.Fatal`.  The difference is that `t.Fatal` will panic and force an immediate exit of the function (any cleanup via defer still gets executed).  `t.Error`, however does not exit and flow will continue onto the next statement.

I think we should swap this function over to using `t.Fatal` throughout.

Reply:
I noticed Fatal will not call defer, but as you mentioned, Fatal is good as it will exit the process.